app/vmselect/promql/transform.go

```diff
@@ -1178,23 +1177,21 @@ func transformRangeQuantile(tfa *transformFuncArg) ([]*timeseries, error) {
 	}
 	phi := phis[0]
 	rvs := args[1]
-	hf := histogram.GetFast()
 	for _, ts := range rvs {
-		hf.Reset()
 		lastIdx := -1
-		values := ts.Values
-		for i, v := range values {
+		originValues := ts.Values
+		var values []float64
```

Comment:
This slice can be re-used if it is declared before the outer loop. This will reduce memory allocations.

Reply:
please see updatea